portal/src/graphql/adminapi/UsersList.tsx

```diff
@@ -33,6 +34,41 @@ interface Props {
   onChangeOffset?: (offset: number) => void;
 }
 
+interface UserInfo {
+  username: string;
+  phone: string;
+  email: string;
+}
+
+function extractUserInfoFromIdentities(
+  identities: UsersListQuery_users_edges_node_identities | null
+): UserInfo {
+  const placeholder = "-";
+  const claimsList: unknown[] = [];
+
+  (identities?.edges ?? []).forEach((edge) => {
+    if (edge?.node?.claims == null) {
+      return;
+    }
+    claimsList.push(edge.node.claims);
+  });
+
+  // TODO: consider update UI design to allow multiple email, username and phone number
+  const email =
+    claimsList.map((claims) => (claims as any).email).filter(Boolean)[0] ||
+    placeholder;
+  const username =
+    claimsList
+      .map((claims) => (claims as any).preferred_name)
```

Comment:
Sorry there's a typo in the claim name in issue. It should be `preferred_username`.

Reply:
updated